docs/framework/guides/simple-plugin-tutorial/abbreviation-plugin-level-2.md

```diff
@@ -0,0 +1,532 @@
+---
+category: simple-plugin
+order: 25
+---
+
+# Abbreviation plugin tutorial - part 2
+
+In this part of the tutorial we'll focus on creating a dialog box, which will get user's input.
+
+We'll pick up where we left off in the first part, so make sure you {@link framework/guides/simple-plugin-tutorial/abbreviation-plugin-level-1 start there}, or grab our starter files for this part.
+
+<info-box>
+As we'll mostly work on the UI, we recommend reading up on our {@link framework/guides/architecture/ui-library UI library} before you start coding.
+</info-box>
+
+If you want to see the final product of this tutorial before you plunge in, check out the [demo](#demo).
+
+## Creating a view
+
+The most important part of the UI for this plugin is a dialog box with a form, which will get us user's input.
+
+### Creating a form view template
+
+Let's start by creating a view with a form. It will include two input fields (for the abbreviation and the title), and the 'submit' and 'cancel' buttons. We'll do it in a separate view. Create a new file `abbreviationview.js` in the `abbreviation/` directory.
+
+Our new `FormView` class will extend the {@link framework/guides/architecture/ui-library#views View} class, so start by importing it from the UI library.
+
+In the `FormView` constructor we'll define a template for our abbreviation form. We need to set a tag of the HTML element, and a couple of its attributes. To make sure our view is focusable, let's add {@link framework/guides/deep-dive/focus-tracking#implementing-focusable-ui-components `tabindex="-1"`}.
+
+We'll also pass the editor's {@link module:utils/locale~Locale} instance to the constructor, so we can localize all our UI components with the help of the {@link module:utils/locale~Locale#t `t()` function}.
+
+```js
+// abbreviation/abbreviationview.js
+
+import View from '@ckeditor/ckeditor5-ui';
+
+export default class FormView extends View {
+	constructor( locale ) {
+		super( locale );
+		const t = locale.t;
+
+		this.setTemplate( {
+			tag: 'form',
+			attributes: {
+				class: [ 'ck', 'ck-abbr-form' ],
+				tabindex: '-1'
+			}
+		} );
+	}
+
+}
+```
+Notice that we added two classes. All UI elements of the editor need to have the `ck` class (unless you want to create your own UI and not use our library). We also created a new class for our form, which we'll use later on to style it.
+
+### Creating input fields
+
+As we have two similar input fields to create and we don't want to repeat ourselves, let's define a method `_createInput()`, which will produce them for us. It will accept the label of our input field.
+
+We'll use {@link module:ui/labeledinput/labeledfieldview~LabeledFieldView `LabeledFieldView`} class, and we'll pass it the `createLabeledInputText()` function as the second argument.
```

Comment:
Maybe we can link to the `createLabeledInputText()` function and mention that this is a helper coming from the UI library that allows creating the input in an easier way?